pygmt/src/which.py

```diff
@@ -6,14 +6,13 @@
     GMTTempFile,
     build_arg_string,
     fmt_docstring,
-    kwargs_to_strings,
+    is_nonstr_iter,
     use_alias,
 )
 
 
 @fmt_docstring
 @use_alias(G="download", V="verbose")
-@kwargs_to_strings(fname="sequence_space")
 def which(fname, **kwargs):
```

Comment:
Reading up on https://peps.python.org/pep-0570/#positional-or-keyword-arguments and https://stackoverflow.com/questions/9450656/positional-argument-vs-keyword-argument (see also Max's previous comment at https://github.com/GenericMappingTools/pygmt/pull/731#discussion_r825477816), should we explicitly mark `fname` as positional_or_keyword like this (unsure if syntax is ok)?

```suggestion
def which(/, fname, *, **kwargs):
```

Or just stick with the current one (which works also I think).

Reply:
Better to open a separate issue so that we can have more discussions about the behavior and also have consistent function definitions in the whole project.

Reply:
On second thought,, maybe we should enforce using keyword-only arguments like so:

```suggestion
def which(*, fname, **kwargs):
```

Then the `gmtwhich [ERROR]` you mentioned at https://github.com/GenericMappingTools/pygmt/issues/2361#issue-1581189232 would turn into a slightly more useful error:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[4], line 1
----> 1 cached_files = which([f"@{fname}" for fname in filenames], download="c")

File ~/pygmt/pygmt/helpers/decorators.py:598, in use_alias.<locals>.alias_decorator.<locals>.new_module(*args, **kwargs)
    591     msg = (
    592         "Parameters 'Y' and 'yshift' are deprecated since v0.8.0. "
    593         "and will be removed in v0.12.0. "
    594         "Use Figure.shift_origin(yshift=...) instead."
    595     )
    596     warnings.warn(msg, category=SyntaxWarning, stacklevel=2)
--> 598 return module_func(*args, **kwargs)

TypeError: which() takes 0 positional arguments but 1 was given
```
Well, actually not that useful (it doesn't say that the `fname` keyword should be used. But something to consider.

Reply:
Instead, I prefer to `def which(fname, *, **kwargs)`. It allows `which("file.txt")` which is much simpler than `which(fname="file.txt")`.

Reply:
Ok, let's go with that. I'll approve this PR, but you can apply the change afterwards.

Reply:
```
def which(fname, *, **kwargs)
```
gives a syntax error:
```
'named arguments must follow bare *
```
See https://stackoverflow.com/questions/14301967/what-does-a-bare-asterisk-do-in-a-parameter-list-what-are-keyword-only-parame.

I'll leave the function definition as it is now.